docs/tutorials/introduction.rst

```diff
@@ -0,0 +1,307 @@
+   :class: seealso
+
+   Our example doesn't make use of all the fields available, so we recommend you
+   check out the `PyPA documentation on project metadata`_.
+
+
+.. admonition:: Writing TOML
+   :class: seealso
+
+   If you are not familiar with the TOML configuration language or need some
+   help, be sure to check out
+   `their documentation <https://toml.io/en/v1.0.0>`__.
+
+
+Generating the distribution artifacts
+=====================================
+
+Now that we have a valid Python project, we can release it.
+
+To release the project we first need to generate the distribution artifacts,
+these are files in a standardized format that Python package installers
+understand. There are two kind of artifacts, `source distributions`_, which are
+commonly referred to as *sdists*, and `binary distributions`_, which use a
+custom format named *wheel*, so they're generally referred to as *wheels*.
+
+
+What are the roles of *sdists* and *wheels*
+-------------------------------------------
+
+
+As you might have figured out by the name, *sdists* contain the source code of
+the project, and *wheels* contain a compiled [#pure-wheels]_ version of the
+project, ready to be copied to the filesystem.
+
+If your project has compiled code, its *wheels* will be platform-dependent, and
+if it has native Python modules, the *wheels* will also be Python version
+specific, as the native Python modules depend on the Python ABI.
+FIXME(this seems *a bit* convoluted, it isn't awful, but I feel like it could
+be simplified or explained in a more natural way)
+
+Strictly speaking, you don't need to distribute *wheels*, however they make the
+user experience much nicer. Unless you have any reason not to, we highly
+recommend you distribute *wheels* for at least the most common systems. When
+*wheels* are not available for a system, the project can still be installed, be
+it needs to be build from the *sdist*, which involves fetching all the build
+dependencies and going through the likely expensive build.
+FIXME(strictly speaking -- this doesn't sound great, how do we highlight the
+importance of building wheels in a better sounding way?)
+
+
+.. [#pure-wheels]
+
+   Projects that don't have any compiled code will have a platform-independent
+   -- *pure* -- wheel.
+
+
+Building the project
+--------------------
+
+To generate the distribution artifacts we will use the `pypa/build`_ tool. It
+will create a temporary virtual environment, install all the required build
+dependencies, and ask ``meson-python`` to build the artifacts.
+
+
+.. code-block:: console
+
+   $ pip install build
+   $ python -m build
+
+
+If the build succeeded, you'll have the binary artifacts in the ``dist`` folder.
+
+
+.. admonition:: Check the pypa/build documentation
+   :class: seealso
+
+   To learn more about `pypa/build`_, you can check
+   `their documentation <https://pypa-build.readthedocs.io/>`__.
+
+
+.. admonition:: Building wheels for many systems
+   :class: tip
+
```

Comment:
I'd say here that for a pure Python package, all one needs is one sdist and one wheel for a release. So we'd be good at this point to upload to PyPI. For a package with an extension module, one needs multiple wheels - they are specific to the OS, hardware, and (unless one uses the stable ABI) the Python minor version. So `.. warning::` this must be done with `cibuildwheel`.